platform/posix/transport/src/sockets_posix.c

```diff
@@ -143,29 +160,90 @@ static SocketStatus_t resolveHostName( const char * pHostName,
+    uint16_t netPort = 0;
+
+    assert( pAddrInfo != NULL );
+    assert( pHostName != NULL );
+    assert( hostNameLength > 0 );
+
+    /* Convert port from host byte order to network byte order. */
+    netPort = htons( port );
+
+    if( pAddrInfo->sa_family == AF_INET )
+    {
+        /* Store IPv4 in string to log. */
+        ( ( struct sockaddr_in * ) pAddrInfo )->sin_port = netPort;
+        addrInfoLength = sizeof( struct sockaddr_in );
+        ( void ) inet_ntop( pAddrInfo->sa_family,
+                            &( ( struct sockaddr_in * ) pAddrInfo )->sin_addr,
+                            resolvedIpAddr,
+                            sizeof( resolvedIpAddr ) );
+    }
+    else
+    {
+        /* Store IPv6 in string to log. */
+        ( ( struct sockaddr_in6 * ) pAddrInfo )->sin6_port = netPort;
+        addrInfoLength = sizeof( struct sockaddr_in6 );
+        ( void ) inet_ntop( pAddrInfo->sa_family,
+                            &( ( struct sockaddr_in6 * ) pAddrInfo )->sin6_addr,
+                            resolvedIpAddr,
+                            sizeof( resolvedIpAddr ) );
+    }
+
+    LogDebug( ( "Attempting to connect to server: Host=%.*s, IP address=%s.",
+                ( int32_t ) hostNameLength,
+                pHostName,
+                resolvedIpAddr ) );
+
+    /* Attempt to connect. */
+    connectStatus = connect( tcpSocket, pAddrInfo, addrInfoLength );
+
+    if( connectStatus == -1 )
+    {
+        LogWarn( ( "Failed to connect to server: Host=%.*s, IP address=%s.",
+                   ( int32_t ) hostNameLength,
+                   pHostName,
+                   resolvedIpAddr ) );
+        close( tcpSocket );
+        returnStatus = SOCKETS_CONNECT_FAILURE;
+    }
+    else
+    {
+        LogDebug( ( "Connected to IP address: %s.",
+                    resolvedIpAddr ) );
+    }
+
+    return returnStatus;
+}
+/*-----------------------------------------------------------*/
+
 static SocketStatus_t attemptConnection( struct addrinfo * pListHead,
                                          const char * pHostName,
                                          size_t hostNameLength,
                                          uint16_t port,
                                          int * pTcpSocket,
                                          int32_t maxAttempts )
 {
-    SocketStatus_t returnStatus = SOCKETS_SUCCESS;
+    SocketStatus_t returnStatus = SOCKETS_CONNECT_FAILURE;
     struct addrinfo * pIndex = NULL;
-    struct sockaddr * pAddrInfo;
-    socklen_t addrInfoLength;
-    uint16_t netPort = 0;
-    int curAttempts = 0, connectStatus = 0;
-    char resolvedIpAddr[ INET6_ADDRSTRLEN ];
+    int curAttempts = 0;
 
     assert( pListHead != NULL );
     assert( pHostName != NULL );
     assert( hostNameLength > 0 );
     assert( pTcpSocket != NULL );
 
-    netPort = htons( port );
-
-    LogDebug( ( "Performing DNS lookup: Host=%.*s.",
+    LogDebug( ( "Attempting to connect to: Host=%.*s.",
```

Comment:
This log seems to almost be the same as the one in connectToAddress. I believe it can just be removed.

Reply:
I think this debug log still adds value as this is the only log which will be printed out before hitting any of the error cases. In ConnectToAddress, it requires a valid socket to be created to get to the debug log. I think this log is still a good addition.  Log level is debug as well.

Reply:
Okay but is there any way we can distinguish both log messages further?
Maybe have one be `Attempting to connect to server` while the other can be `Attempting to connect to resolved IP from server`.

Reply:
Yea. Good suggestion. Will update the log in connectToSrever.